### DISCUSSION TITLE
be able to use custom clock in QUILL_LOGGER_CALL_LIMIT

### DISCUSSION THREAD

**Discussion by jol** [odygrd/quill#628] [General]
Hi,

The feature about LOGGER_CALL_LIMIT is nice. However, in some cases, it could be more usable. I have a system used for production (live time) or backtesting (controlled time). In the case of backtesting, I use a custom time that I advance when needed, so in this case, I cannot use code based on QUILL_LOGGER_CALL_LIMIT. It uses:

```cpp
auto const now = std::chrono::steady_clock::now();  
```

So, I would like to know if it would be possible to use a time provider (the one defined) in the  LOGGER_CALL_LIMIT implementation.

Kind regards,

Joël

**Comment by odygrd** [comment]
Hey, thanks for the suggestion! Currently, this feature isn’t available, but it’s something I can consider adding in the future. The main challenge would be the added complexity with macros, as we’re already have quite a few.

Another thing to keep in mind is that if you’re calling this macro inside a main loop, that loop could still be active in both live and simulated environments. In that case, you’d likely want to always use a live clock rather than a simulated one.

As a workaround for now, I’d recommend creating your own custom macro based on the `QUILL_LOGGER_CALL_LIMIT` implementation. You could then pass in either a live or simulated clock as an argument to your custom macro.
